test/ci/analysis.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+set -ve
+
+MACHINE=$(uname -m)
+uname -a
+gcc --version
+clang --version
+
+echo "::group::analyze"
+# Disable inline assembly for analyze
+if [ "${MACHINE}" == "x86_64" ]; then
+	export SSSE3_CFLAGS="-mssse3 -DBASE64_SSSE3_USE_ASM=0"
+	export SSE41_CFLAGS="-msse4.1 -DBASE64_SSE41_USE_ASM=0"
+	export SSE42_CFLAGS="-msse4.2 -DBASE64_SSE42_USE_ASM=0"
+	export AVX_CFLAGS="-mavx -DBASE64_AVX_USE_ASM=0"
+	export AVX2_CFLAGS="-mavx2 -DBASE64_AVX2_USE_ASM=0"
+	# Temporarily disable AVX512; it is not available in CI yet.
+	# export AVX512_CFLAGS="-mavx512vl -mavx512vbmi"
+elif [ "${MACHINE}" == "aarch64" ]; then
+	export NEON64_CFLAGS="-march=armv8-a"
+elif [ "${MACHINE}" == "armv7l" ]; then
+	export NEON32_CFLAGS="-march=armv7-a -mfloat-abi=hard -mfpu=neon"
+fi
+make analyze
+echo "::endgroup::"
+
+echo "::group::valgrind (no assembly)"
+# Disable inline assembly for this run
+if [ "${MACHINE}" == "x86_64" ]; then
+	export SSSE3_CFLAGS="-mssse3 -DBASE64_SSSE3_USE_ASM=0"
+	export SSE41_CFLAGS="-msse4.1 -DBASE64_SSE41_USE_ASM=0"
+	export SSE42_CFLAGS="-msse4.2 -DBASE64_SSE42_USE_ASM=0"
+	export AVX_CFLAGS="-mavx -DBASE64_AVX_USE_ASM=0"
+	export AVX2_CFLAGS="-mavx2 -DBASE64_AVX2_USE_ASM=0"
+	# Temporarily disable AVX512; it is not available in CI yet.
+	# export AVX512_CFLAGS="-mavx512vl -mavx512vbmi"
+elif [ "${MACHINE}" == "aarch64" ]; then
+	export NEON64_CFLAGS="-march=armv8-a"
+elif [ "${MACHINE}" == "armv7l" ]; then
+	export NEON32_CFLAGS="-march=armv7-a -mfloat-abi=hard -mfpu=neon"
+fi
+make clean
+make
+make -C test valgrind
+echo "::endgroup::"
+
+echo "::group::valgrind (assembly)"
+# Force inline assembly
+if [ "${MACHINE}" == "x86_64" ]; then
+	export SSSE3_CFLAGS="-mssse3 -DBASE64_SSSE3_USE_ASM=1"
+	export SSE41_CFLAGS="-msse4.1 -DBASE64_SSE41_USE_ASM=1"
+	export SSE42_CFLAGS="-msse4.2 -DBASE64_SSE42_USE_ASM=1"
+	export AVX_CFLAGS="-mavx -DBASE64_AVX_USE_ASM=1"
+	export AVX2_CFLAGS="-mavx2 -DBASE64_AVX2_USE_ASM=1"
+	# Temporarily disable AVX512; it is not available in CI yet.
+	# export AVX512_CFLAGS="-mavx512vl -mavx512vbmi"
+elif [ "${MACHINE}" == "aarch64" ]; then
+	export NEON64_CFLAGS="-march=armv8-a"
+elif [ "${MACHINE}" == "armv7l" ]; then
+	export NEON32_CFLAGS="-march=armv7-a -mfloat-abi=hard -mfpu=neon"
+fi
+make clean
+make
+make -C test valgrind
+echo "::endgroup::"
```

Comment:
There's a lot of repetition in this file. I'd be tempted to avoid the copypasting and use a shell loop to enable/disable inline asm:

```sh
for ASM_ENABLED in 0 1; do
    ...
done
```

Reply:
yes indeed